### PR TITLE
refactored logging into LogUtil, this is a more robust solution than …

### LINKED ISSUE #46: Logging more elegantly
# Problem
Differentiating between logging in production and debug currently requires to prepend the log statement with `if (BuildConfig.DEBUG)`. It would be desirable to remove log statements in all other build types than debug in a more meta-way so that you don't have to prepend the if statement every time.

# Solution
A possible solution would be to use proguard rules. [Proguard for libraries](https://developer.android.com/studio/projects/android-library) seems to be a perfect fit. So far, creating an aar with a `proguard.txt` file in it with specified rules has been done. The rules aren't respected upon testing it in another app however. Therefore, thoroughly test this functionality to the point of production releases. A possible scenario for this could be:

1. Make appropriate changes to gradle build files and/or proguard rules.
1. Generate a .aar file from the SDK.
1. Use the .aar file in another project.
1. Build and install a signed .apk file from the other project, which implements the .aar.
1. Run the app. There should be no logging messages from the SDK in the console.

> Optional (but useful): Also generate a signed .apk with our debug build type. Verify that the logs will appear in this build.

### PR DESCRIPTION
**Pull request by EdwinVanRooij**
…previously, where you might forget the if statement. Use the log util now. closes #46

### CODE REVIEW

File: android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ *
+ * This way, we can use the Android Studio search tool to make sure there is only one 'Log.e(...)'
+ * statement in the entire project.
+ *
+ * Suggestion: It would be better to use ProGuard to remove logs in the release build variant.
+ * This, however, has proven to require a lot of understanding about ProGuard and thus time consuming.
+ */
+public class LogUtil {
+
+  public static void LogE(final String TAG, String message) {
```
Comment (Roconda) [inline android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java]: add javadoc

File: android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    if (BuildConfig.DEBUG) {
+      Log.e(TAG, message);
+    }
+  }
+
+  public static void LogE(final String TAG, String message, Throwable tr) {
```
Comment (Roconda) [inline android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java]: add javadoc

File: android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    if (BuildConfig.DEBUG) {
+      Log.e(TAG, message, tr);
+    }
+  }
+
+  public static void LogW(final String TAG, String message) {
```
Comment (Roconda) [inline android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java]: add javadoc

File: android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    if (BuildConfig.DEBUG) {
+      Log.w(TAG, message);
+    }
+  }
+
+  public static void LogW(final String TAG, String message, Throwable tr) {
```
Comment (Roconda) [inline android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java]: add javadoc

File: android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    if (BuildConfig.DEBUG) {
+      Log.w(TAG, message, tr);
+    }
+  }
+
+  public static void LogI(final String TAG, String message) {
```
Comment (Roconda) [inline android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java]: add javadoc

File: android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    if (BuildConfig.DEBUG) {
+      Log.i(TAG, message);
+    }
+  }
+
+  public static void LogI(final String TAG, String message, Throwable tr) {
```
Comment (Roconda) [inline android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java]: add javadoc

File: android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    if (BuildConfig.DEBUG) {
+      Log.i(TAG, message, tr);
+    }
+  }
+
+  public static void LogD(final String TAG, String message) {
```
Comment (Roconda) [inline android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java]: add javadoc

File: android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    if (BuildConfig.DEBUG) {
+      Log.d(TAG, message);
+    }
+  }
+
+  public static void LogD(final String TAG, String message, Throwable tr) {
```
Comment (Roconda) [inline android/sdk/src/main/java/io/o2mc/sdk/util/LogUtil.java]: add javadoc

**Comment by EdwinVanRooij** [comment]
I deliberately omitted javadoc per method, to instead keep it all in the description of the class. 
This way I avoid repeating myself (thus avoiding duplicate and useless extra lines). 

I feel like
```java
/**
 * This class serves as a wrapper around the android.util.Log class.
 * It makes sure logs are only generated in the debug build variant.
 *
 * Putting the checks in this class instead of before every log individually reduces human-errors
 * (i.e. you might forget to add a check somewhere).
 *
 * This way, we can use the Android Studio search tool to make sure there is only one 'Log.e(...)'
 * statement in the entire project.
 *
 * Suggestion: It would be better to use ProGuard to remove logs in the release build variant.
 * This, however, has proven to require a lot of understanding about ProGuard and thus time consuming.
 */
```
Is descriptive enough for the entire class.
However, I added a suppress warnings for javadocs on methods to explain this in the code as well.

**Comment by Roconda** [comment]
Fair enough since it's internal code. Approving this PR.

**Pull request merged**